engine/src/main/java/org/hibernate/validator/constraints/BitcoinAddressType.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.constraints;
+
+public enum BitcoinAddressType {
+	ANY("Bitcoin", null),
+	P2PKH("Legacy (P2PKH)", "^(1)[a-zA-HJ-NP-Z0-9]{25,61}$"),
+	P2SH("Nested SegWit (P2SH)", "^(3)[a-zA-HJ-NP-Z0-9]{33}$"),
+	BECH32("Native SegWit (Bech32)", "^(bc1)[a-zA-HJ-NP-Z0-9]{39,59}$"),
+	P2WSH("SegWit variant of P2SH (P2WSH)", "^(bc1q)[a-zA-HJ-NP-Z0-9]{58}$"),
+	P2WPKH("SegWit variant of P2PKH (P2WPKH)", "^(bc1q)[a-zA-HJ-NP-Z0-9]{38}$"),
+	P2TR("Taproot (P2TR)", "^(bc1p)[a-zA-HJ-NP-Z0-9]{58}$");
```

Comment:
BTW I see that this description is used in the validation message. But as they use regular words they may need to be translated as well... So I'd suggest exploring an option of placing the description in the translation file as well and then let EL do the string interpolation.

Reply:
Done. When you have a chance to take a look please :)